### PR TITLE
Fix some more assembly versioning issues

### PR DESCRIPTION
**Pull request by Therzok**

### CODE REVIEW

File: main/src/core/MonoDevelop.Ide/packages.config [hunk 1]
```diff
@@ -11,7 +11,7 @@
   <package id="System.Collections.Immutable" version="1.3.1" targetFramework="net45" />
   <package id="System.Composition" version="1.0.31" targetFramework="net45" />
   <package id="System.IO.Compression" version="4.3.0" targetFramework="net461" />
-  <package id="System.Threading.Tasks.DataFlow" version="4.7.0" targetFramework="net45" />
+  <package id="System.Threading.Tasks.DataFlow" version="4.5.24" targetFramework="net45" />
```
Comment (KirillOsenkov) [inline main/src/core/MonoDevelop.Ide/packages.config]: Don't we rather want to redirect up to 4.7.0?

**Review by KirillOsenkov — APPROVED** [review]

File: main/src/core/MonoDevelop.Ide/packages.config (hunk 1 above)
Comment (Therzok) [reply main/src/core/MonoDevelop.Ide/packages.config]: We could, I just looked at the easiest way to get rid of the conflict in the log. This is what roslyn uses, we can go up if we want.

**Review by mrward — APPROVED** [review]

File: main/src/core/MonoDevelop.Core/packages.config [hunk 2]
```diff
@@ -50,12 +50,12 @@
...
-  <package id="System.Composition.Convention" version="1.0.31" targetFramework="net461" />
-  <package id="System.Composition.Hosting" version="1.0.31" targetFramework="net461" />
-  <package id="System.Composition.Runtime" version="1.0.31" targetFramework="net461" />
-  <package id="System.Composition.TypedParts" version="1.0.31" targetFramework="net461" />
+  <package id="System.Composition" version="1.0.31" targetFramework="net45" />
+  <package id="System.Composition.AttributedModel" version="1.0.31" targetFramework="net45" />
+  <package id="System.Composition.Convention" version="1.0.31" targetFramework="net45" />
+  <package id="System.Composition.Hosting" version="1.0.31" targetFramework="net45" />
+  <package id="System.Composition.Runtime" version="1.0.31" targetFramework="net45" />
+  <package id="System.Composition.TypedParts" version="1.0.31" targetFramework="net45" />
```
[the remark this answers is not in the document]
Comment (Therzok) [reply main/src/core/MonoDevelop.Core/packages.config]: From what I've seen, net461 fallsback to net45. I can modify it to be net461 👍

[the remark this answers is not in the document]
Comment (mrward) [reply main/src/core/MonoDevelop.Core/packages.config]: Not sure it is a problem - just seemed different to the rest :)

File: main/src/core/MonoDevelop.Ide/packages.config (hunk 1 above)
Comment (Therzok) [reply main/src/core/MonoDevelop.Ide/packages.config]: Trying that!

**Pull request merged**